sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -589,22 +593,41 @@ case class DateFormatClass(left: Expression, right: Expression, timeZoneId: Opti
 
   override def inputTypes: Seq[AbstractDataType] = Seq(TimestampType, StringType)
 
-  override def withTimeZone(timeZoneId: String): TimeZoneAwareExpression =
-    copy(timeZoneId = Option(timeZoneId))
+  override def withTimeZone(timeZoneId: String): TimeZoneAwareExpression = {
+    val tf = if (formatter.isEmpty && right.foldable) {
+      val format = right.eval().toString
+      Some(TimestampFormatter(
+        format,
+        DateTimeUtils.getZoneId(timeZoneId),
+        Locale.US))
```

Comment:
If there is no other reasons, shall we use a single line instead of four lines?
```scala
Some(TimestampFormatter(format, DateTimeUtils.getZoneId(timeZoneId)))
```

Reply:
Replaced by the single line